inst/templates/ad_advs.R

```diff
@@ -0,0 +1,271 @@
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Anthropometric measurement", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Anthropometric measurement", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Anthropometric measurement", 1,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 4, "Anthropometric measurement", 1,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5, "Anthropometric measurement", 1,
+  "DIABP", "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "Vital Sign", 2,
+  "PULSE", "PULSE", "Pulse Rate (beats/min)", 7, "Vital Sign", 2,
+  "SYSBP", "SYSBP", "Systolic Blood Pressure (mmHg)", 8, "Vital Sign", 2,
+  "TEMP", "TEMP", "Temperature (C)", 9, "Vital Sign", 2
+)
+
+# Add parameter (PARAMCD) info to enable later ADVS derivations. Additional
+# parameter information will be merged again, after all AVDS derivations are
+# completed.
+
+# See function documentation for `derive_vars_merged_lookup()`:
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html)
+advs <- advs %>%
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  )
+
+
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add vital sign analysis date (ADT) and treatment start date (TRTSDT)
+advs <- advs %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = VSDTC)
+advs <- advs %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+
+# Derive visit info ----
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis time point (ATPT, ATPTN) and analysis visit (AVISIT, AVISITN)
+advs <- advs %>%
+  mutate(
+    ATPT = VSTPT,
+    ATPTN = VSTPTNUM,
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      VISIT == "BASELINE" ~ 0,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
+  )
+
+
+# Derive results ----
+# See the "Derive Results (AVAL, AVALC)" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#aval)
+
+# Derive analysis result (AVAL)
+advs <- advs %>%
+  mutate(AVAL = VSSTRESN)
+
+
+# Derive domain specific variables ----
+# See the "Derive Additional Parameters" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
+
+# Derive BMI
+advs <- advs %>%
+  filter(VSTESTCD != "BMI") %>%
+  derive_param_bmi(
+    by_vars = exprs(
+      STUDYID, USUBJID, !!!adsl_vars,
+      AVISITN, ADT, ADY, ATPT, ATPTN
```

Comment:
```suggestion
      AVISIT, AVISITN, ADT, ADY, ATPT, ATPTN
```

`AVISIT` never gets populated if not in `by_vars`.